internal/controller/pipelinerollout/pipelinerollout_progressive.go

```diff
@@ -403,26 +406,13 @@ func scalePipelineVerticesToZero(
 		}
 	}
 	if !allVerticesScaledDown {
-
-		vertices, _, err := unstructured.NestedSlice(pipelineDef.Object, "spec", "vertices")
-		if err != nil {
-			return fmt.Errorf("error while getting vertices of promoted pipeline: %w", err)
-		}
-		for _, vertex := range vertices {
-			if vertexAsMap, ok := vertex.(map[string]any); ok {
-
-				if err := unstructured.SetNestedField(vertexAsMap, int64(0), "scale", "min"); err != nil {
-					return err
-				}
-
-				if err := unstructured.SetNestedField(vertexAsMap, int64(0), "scale", "max"); err != nil {
-					return err
-				}
-			}
+		zero := int64(0)
+		for i := range vertexScaleDefinitions {
+			vertexScaleDefinitions[i].ScaleDefinition = &apiv1.ScaleDefinition{Min: &zero, Max: &zero}
 		}
 
 		numaLogger.Debug("Scaling down all vertices to 0 Pods")
-		if err := patchPipelineVertices(ctx, pipelineDef, vertices, c); err != nil {
+		if err := applyScaleValuesToLivePipeline(ctx, pipelineDef, vertexScaleDefinitions, c); err != nil {
```

Comment:
this call just patches in the new min and max as opposed to patching all fields of the vertices